subcommand/controller/command.go

```diff
@@ -103,6 +99,12 @@ func (c *Command) run() error {
 	// Set up ECS client.
 	ecsClient := ecs.New(clientSession)
 
+	// Token containing `acl:write`, `operator:write` and `node:write` privileges against the server
+	token := config.GetConsulToken()
+	if token != "" {
+		return fmt.Errorf("CONSUL_HTTP_TOKEN should be non empty")
```

Comment:
I don't think we should make the token mandatory (unfortunately). If Consul does not have ACLs enabled, then the controller does not need an ACL token to run. It would need to deregister agentless services, but would not need to cleanup ACL tokens.

Reply:
Cool. In that case, we could make this token optional. If a token is present we use it to login. If not, we login via the auth method configuration provided via the `consulLogin` field.